contracts/modules/token/TokenFunctions.sol

```diff
@@ -112,26 +102,33 @@ contract TokenFunctions is Iupgradable {
    * @return the status of the successful burning
    */
   function burnDepositCN(uint coverId) public onlyInternal returns (bool success) {
+
     address _of = qd.getCoverMemberAddress(coverId);
-    uint amount;
-    (amount,) = td.depositedCN(coverId);
-    amount = (amount.mul(50)).div(100);
     bytes32 reason = keccak256(abi.encodePacked("CN", _of, coverId));
-    tc.burnLockedTokens(_of, reason, amount);
-    success = true;
+    uint lockedAmount = tc.tokensLocked(_of, reason);
+
+    (uint amount,) = td.depositedCN(coverId);
+    amount = amount.div(2);
+
+    // limit burn amount to actual amount locked
+    uint burnAmount = lockedAmount < amount ? lockedAmount : amount;
```

Comment:
is this necessary? suppose it's the first claim, this doesn't apply. suppose it's the 2nd one, then amount.div(2)  would still yield smth that is <= lockedAmount

only way this logic executes is if you're attempting a 3rd burn, do you forsee that happening somehow